sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala

```diff
@@ -339,6 +340,39 @@ object EliminateDistinct extends Rule[LogicalPlan] {
   }
 }
 
+/**
+ * Remove useless FILTER clause for aggregate expressions.
+ * This rule should be applied before RewriteDistinctAggregates.
+ */
+object EliminateAggregateFilter extends Rule[LogicalPlan] {
+  override def apply(plan: LogicalPlan): LogicalPlan = plan transformExpressions  {
+    case ae @ AggregateExpression(_, _, _, Some(Literal.TrueLiteral), _) =>
+      ae.copy(filter = None)
+    case ae @ AggregateExpression(af, _, _, Some(Literal.FalseLiteral), _) =>
+      rewrite(ae, af)
+    case ae @ AggregateExpression(af, _, _, Some(e), _)
```

Comment:
We don't need to handle this. We should assume `ConstantFolding` will take care of it.

Reply:
OK